### PR TITLE
Patterns: Add a curation filter to the UI

### LINKED ISSUE #580: Add select for filtering the pattern directory
Consolidating next steps for https://github.com/WordPress/pattern-directory/issues/395 and https://github.com/WordPress/pattern-directory/issues/542. Let's add a select control for filtering pattern results at a high level. 

This will introduce "Curated" and "Community" top-level filters, where "Curated" patterns are hand-picked to be surfaced front-and-center on the Patterns Directory, and eventually within the Pattern Explorer via https://github.com/WordPress/gutenberg/issues/44611. 

### Tasks 

- [ ] Add a filtering select with the options "Curated" and "Community" above the pattern categories.
- [ ] Move the sort select beside the new filter.
- [ ] Remove context bar in lieu of simpler pattern count to the right of the filters.
- [ ] Set "Curated" as the default value.
- [ ] Improve design metrics re the visuals below (could be a follow-up)

### Visuals

patterns

### PR DESCRIPTION
**Pull request by ryelle**
This adds a new filter to the UI for "curation", as described in #580— "curated" patterns are those which have been picked to auto-load into core. The new filter is a dropdown with "curated" and "community", which lets the viewer switch between those  two views. It defaults to curated on home & category archives, and uses curated for the empty state recommendations.

The author archive, favorites, and my patterns don't have this filter, and default to showing "all" patterns.

Fixes #580

Note, this is based off #583, which adds the parameter to the API endpoint.

### Screenshots

I applied this to my sandbox to take screenshots, so this is using production patterns.

|  | Curated | Community |
|---|---|---|
| Home, default sort | home-curated | home-community |
| Home, sorted by popular | home-curated-popular | home-community-popular |
| Category, default sort | category-curated | category-community |

<details>
<summary>Favorites & My Patterns have no changes, but screenshots for reference</summary>

favorites

my-patterns

</details>

On medium-smaller screens, the filters wrap, then at 782px the menu collapses.

This new change means that the Call to Action menu item defaults to empty (https://wordpress.org/patterns/categories/call-to-action/), but you can see here the empty state also uses the "curated" patterns.

home-curated-cta-empty

### How to test the changes in this Pull Request:

1. Check out the branch and build the project
2. Make sure you have some patterns with the `core` keyword: `/wp-admin/edit.php?wporg-pattern-keyword=core&post_type=wporg-pattern&post_parent=0&post_status=publish` — if not, you should be able to add some via wp-admin
3. View the homepage, it should default to the core patterns
4. Try using the filters on the UI to switch to community
5. Navigate through categories, it should persist your chosen curation
6. My Patterns, My Favorites, and author archives should still work, without a curation filter

### CODE REVIEW

**Comment by jasmussen** [comment]
Looks good to me!

**Comment by fcoveram** [comment]
Same. It looks good 👍

**Review by outdoor2kode — APPROVED** [review]
Tested on sandbox and worked as described. Codes-wise look good to me as well 👍 

> This new change means that the Call to Action menu item defaults to empty (https://wordpress.org/patterns/categories/call-to-action/), but you can see here the empty state also uses the "curated" patterns.

Does it need to make any changes to it accordingly, maybe rephrase the text under `No results found`? Any better ways to handle this?

**Comment by StevenDufresne** [comment]
I probably missed this in the [original issue](https://github.com/WordPress/pattern-directory/issues/580), but does this mean I can never actually see "all" the patterns? I have to either subsection them by "community/curated"? 

What if I don't care about whether they are community or curated and just want them all?

**Comment by annezazu** [comment]
Good point. I can see where folks would want to see all and we could still prioritize by curated. What do you think @richtabor? This looks good as an initial iteration though.

**Comment by jasmussen** [comment]
The missing "all" is very much intentional, and an attempt to bring a little structure to what is currently an unfortunate kitchen sink. I like to think that as we roll out potential features in the future like author patterns, pattern bundles, clearer guidelines for each category, related patterns etc, we can see what that does to the curation and then revisit.

**Comment by ryelle** [comment]
Technically, we do have the ability to show `all`, just no UI to trigger it. It's not prioritized by curated (because that introduces a new sort order variable), but it does exist. In case we want to bring it back (now or later).
